interp/src/debugger/debugger_core.rs

```diff
@@ -113,7 +114,29 @@ impl<C: AsRef<Context> + Clone> Debugger<C> {
             status: self
                 .interpreter
                 .get_currently_running_groups()
-                .map(|x| self.program_context.as_ref().lookup_name(x).clone())
+                .map(|x| {
+                    let group_name =
+                        self.program_context.as_ref().lookup_name(x).clone();
+                    let parent_comp = self
+                        .program_context
+                        .as_ref()
+                        .primary
+                        .components
+                        .keys()
+                        .find(|comp_id| {
+                            self.program_context.as_ref().secondary[*comp_id]
+                                .definitions
+                                .groups()
+                                .contains(x)
+                        })
+                        .unwrap();
+                    let parent_name = self
+                        .program_context
+                        .as_ref()
+                        .lookup_name(parent_comp)
+                        .clone();
```

Comment:
You can use the more recently added `get_component_from_group`!
https://github.com/calyxir/calyx/blob/0d578c770a0a34d2d886b83d286ea3ac1498f3cb/interp/src/flatten/structures/context.rs#L345

You'll need to merge in the more recent changes but should simplify things
somewhat